### ISSUE TITLE
tModLoader does not default to Mono/.NET 4.0/4.5 when compiling for OSX/Linux platforms

### ISSUE THREAD

**Issue by gamrguy** [bluemagic123/tModLoader#156]
### Description

When building a mod through tModLoader, tML uses the latest installed version of Mono/.NET as the target framework version. However, OSX/Linux platforms are stuck with Mono 4.0, while Windows has .NET 4.6+. This causes crashes on attempting to load any mod on OSX/Linux that uses certain features of C#, namely extensions and possibly others.

### Steps to Reproduce

1. Build any mod that contains an extension on Windows through tModLoader, or manually build the mod with a .NET version greater than 4.5
2. Attempt to load the mod on OSX/Linux, or potentially a Windows Terraria being forced to use .NET 4.0
3. Observe crash.

### Expected behavior 

tModLoader should recognize that OSX/Linux only has Mono 4.0 and should therefore compile for Mono with that as the target framework version.

### Actual behavior

tModLoader completely ignores that OSX/Linux only has Mono 4.0 and instead compiles for .NET 4.5+ (whatever is installed on the system).

### Reproduction frequency

100%.

### Versions

All versions of tModLoader that have ever existed since the release of .NET versions greater than 4.5, presumably.

### Mods

Issue is within tModLoader's compiling code itself, not with mods installed. That said, crashes will only occur on OSX/Linux if specific C# features are used in the compiled mod, of which extensions are currently the only confirmed feature to do so.

### Additional Information

Manually compiling any mod that exhibits the crash targeting Mono/.NET 4.0 results in fixing the crash on OSX/Linux platforms. The effects of this on Windows platforms are untested, though no adverse effects are expected provided that only the Mono-compatible build is affected.

Building for either 4.0 or 4.5 works as expected; these are indicated on my IDE with `Mono/.NET` before the version number. Anything higher only says `.NET` and triggers the crash.

**Comment by Chicken-Bones** [comment]
Unfortunately it's very difficult to find the right .Net 4.0 assemblies to compile with. Generally they're stored in Program Files (x86)/ReferenceAssemblies/... but that might only be if the 4.0 framework was installed before it was upgraded, not if a later version was installed directly. I expect Visual Studio and other .NET framework development platforms to include the files required for targetting all versions of the framework, but there's no guarantee such an environment is on a user's system.

Long story short, we decided to fix the symptom instead of the problem d445b9bd1163e6ac62e765d94a7da922882a88b8

**Comment by Chicken-Bones** [comment]
It appears that we've now fixed the problem (mods are always built against .NET 4.5 now)
I'm commenting here incase the issue resurfaces somehow, though I've been unable to replicate it
